### PR TITLE
Add deploy command doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
Adding some more detail to the API doc for the new `ng deploy` command.

### CODE REVIEW

File: packages/angular/cli/commands/deploy-long.md [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
 It takes an optional project name, as specified in the `projects` section of the `angular.json` workspace configuration file.
+When a project name is not supplied, executes the `deploy` builder for the default project.
 
-When a project name is not supplied, the CLI will execute the `deploy` builder for the default project.
+The deploy builder defined in a deployment platform package is automatically added to a project's configuration when you add the package to the project.
+For example:
+
+```json
+"projects": {
+     {"my-project": {
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]:
```suggestion
     "my-project": {
```

File: packages/angular/cli/commands/deploy-long.md [hunk 2]
```diff
@@ -1,9 +1,38 @@
...
+         ...
+        {"architect": {
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]:
```suggestion
        "architect": {
```

File: packages/angular/cli/commands/deploy-long.md [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+            ...
+            "deploy": {
+                "builder": "@angular/fire:deploy",
+                "options": {}
+                    }
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]:
```suggestion
```

File: packages/angular/cli/commands/deploy-long.md [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
+                }
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]:
```suggestion
```

File: packages/angular/cli/commands/deploy-long.md [hunk 5]
```diff
@@ -1,9 +1,38 @@
...
+            }
+        }
+    }
+ }
+ ```
+
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]: I think we can trim the rest of the file.

File: packages/angular/cli/commands/deploy-long.md [hunk 6]
```diff
@@ -1,9 +1,38 @@
-The `ng deploy` command is a shortcut for:
+The command is a shortcut for:
 
-```
+<code-example language="bash">
 ng run [PROJECT_NAME]:deploy
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]: This is now just `ng deploy [PROJECT_NAME]`

File: packages/angular/cli/commands/deploy-long.md [hunk 7]
```diff
@@ -1,9 +1,38 @@
...
-```
+</code-example>
 
 It takes an optional project name, as specified in the `projects` section of the `angular.json` workspace configuration file.
+When a project name is not supplied, executes the `deploy` builder for the default project.
 
-When a project name is not supplied, the CLI will execute the `deploy` builder for the default project.
+The deploy builder defined in a deployment platform package is automatically added to a project's configuration when you add the package to the project.
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]: Can we update it to something like:

To use the `ng deploy` command `ng add` a package that implements deployment capabilities to your favorite platform. The `ng-add` schematics of the package will automatically update your workspace configuration, adding a deployment CLI builder.

Comment (jbogarthyde) [reply packages/angular/cli/commands/deploy-long.md]: Also adding link to CLI Builder guide.

File: packages/angular/cli/commands/deploy-long.md [hunk 8]
```diff
@@ -1,9 +1,22 @@
-The `ng deploy` command is a shortcut for:
+The command takes an optional project name, as specified in the `projects` section of the `angular.json` workspace configuration file.
+When a project name is not supplied, executes the `deploy` builder for the default project.
 
-```
-ng run [PROJECT_NAME]:deploy
-```
+To use the `ng deploy` command, use `ng add` to add a package that implements deployment capabilities to your favorite platform.
+Adding the package automatically updates your workspace configuration, adding a deployment
+`CLI builder`(guide/cli-builder).
```
Comment (mgechev) [inline packages/angular/cli/commands/deploy-long.md]: Users see `deploy-long.md` when the run `ng deploy --help` as well and this won't be rendered properly. Maybe we can drop it?

```suggestion
`CLI builder`.
```

**Comment by mgechev** [comment]
@kyliau the failure is because the CLI's size increased with the doc change.

File: packages/angular/cli/commands/deploy-long.md (hunk 8 above)
Comment (jbogarthyde) [reply packages/angular/cli/commands/deploy-long.md]: OK -- but I think the long description only shows up in AIO, not in the help.

Comment (mgechev) [reply packages/angular/cli/commands/deploy-long.md]: Yep, feel free to ignore my comment.

**Pull request merged**